core/ledger/kvledger/snapshot.go

```diff
@@ -273,15 +273,28 @@ func (p *Provider) CreateFromSnapshot(snapshotDir string) (ledger.PeerLedger, st
 			errors.WithMessage(err, "error while importing data into config history Mgr"),
 		)
 	}
+
+	configHistoryRetiever := p.configHistoryMgr.GetRetriever(ledgerID)
 	btlPolicy := pvtdatapolicy.ConstructBTLPolicy(
 		&mostRecentCollectionConfigFetcher{
 			DeployedChaincodeInfoProvider: p.initializer.DeployedChaincodeInfoProvider,
-			Retriever:                     p.configHistoryMgr.GetRetriever(ledgerID),
+			Retriever:                     configHistoryRetiever,
 		},
 	)
 	purgeMgrBuilder := pvtstatepurgemgmt.NewPurgeMgrBuilder(ledgerID, btlPolicy, p.bookkeepingProvider)
 
-	if err = p.dbProvider.ImportFromSnapshot(ledgerID, savepoint, snapshotDir, purgeMgrBuilder); err != nil {
+	snapshotDataImporter, err := p.pvtdataStoreProvider.SnapshotDataImporterFor(
```

Comment:
Nit. Similar to line 284, we could have a `NewPvtDataMissingInfoBuilder()` here to be explicit about what is being stored in the pvtdatastore during the import phase.